prebuilt-tasks/src/test/java/com/redhat/parodos/tasks/git/GitPushTaskTest.java

```diff
@@ -0,0 +1,238 @@
+package com.redhat.parodos.tasks.git;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Stream;
+
+import com.redhat.parodos.workflow.exception.MissingParameterException;
+import com.redhat.parodos.workflow.parameter.WorkParameter;
+import com.redhat.parodos.workflow.utils.WorkContextUtils;
+import com.redhat.parodos.workflows.work.WorkContext;
+import com.redhat.parodos.workflows.work.WorkReport;
+import com.redhat.parodos.workflows.work.WorkStatus;
+import junit.framework.TestCase;
+import lombok.extern.slf4j.Slf4j;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.InitCommand;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.lib.ObjectId;
+import org.eclipse.jgit.lib.Ref;
+import org.eclipse.jgit.lib.Repository;
+import org.eclipse.jgit.revwalk.RevCommit;
+import org.eclipse.jgit.revwalk.RevWalk;
+import org.eclipse.jgit.transport.URIish;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+
+@Slf4j
+public class GitPushTaskTest extends TestCase {
+
+	private GitPushTask task;
+
+	private Path gitRepoPath;
+
+	private Path tempDir;
+
+	private Repository repository;
+
+	private Path remotePath;
+
+	private Repository remoteRepository;
+
+	@BeforeEach
+	public void setUp() throws Exception {
+		task = new GitPushTask();
+		task.setBeanName("git-push-task");
+
+		tempDir = Files.createTempDirectory("git-repo");
+		gitRepoPath = tempDir.resolve(GitConstants.GIT_FOLDER);
+		repository = initRepo(gitRepoPath);
+
+		remotePath = Files.createTempDirectory("git-remote");
+		remoteRepository = initRepo(remotePath.resolve(GitConstants.GIT_FOLDER));
+		log.info("Created a new repository at {} ", this.repository.getDirectory());
+	}
+
+	@AfterEach
+	public void tearDown() throws Exception {
+		assertDoesNotThrow(() -> {
+			repository.close();
+		});
+		try (Stream<Path> walk = Files.walk(tempDir)) {
+			walk.sorted(java.util.Comparator.reverseOrder()).map(Path::toFile).forEach(p -> {
+				assertDoesNotThrow(() -> Files.delete(p.toPath()));
+			});
+		}
+	}
+
+	@Test
+	public void testParams() {
+		// when
+		List<WorkParameter> params = task.getWorkFlowTaskParameters();
+
+		// then
+		assertThat(params).isNotNull();
+		assertThat(params.size()).isEqualTo(2);
+		assertThat(params.get(0).getKey()).isEqualTo("path");
```

Comment:
I'd replace this with so we won't be too opinionated about the position of parameters (despite the fact the type is List and not Set)
```
assertThat(params).contains("path", "repo");
```